src/lib/__tests__/InstantSearch-integration-test.ts

```diff
@@ -44,3 +46,72 @@ describe('configure', () => {
     });
   });
 });
+
+describe('middleware', () => {
+  it("runs middlewares' onStateChange uiState has changed", async () => {
```

Comment:
I think that's what you meant?

```suggestion
  it("runs middlewares' onStateChange when uiState changes", async () => {
```